snapcraft/internal/meta/snap.py

```diff
@@ -317,6 +319,31 @@ def from_dict(cls, snap_dict: Dict[str, Any]) -> "Snap":
 
         layout = snap_dict.pop("layout", None)
         license = snap_dict.pop("license", None)
+
+        # snap.yaml will have links, snapcraft.yaml the top level entries.
+        # some considerations:
+        # - links should not be allowed in snapcraft.yaml by the schema.
+        # - the top level entries would never exist in snap.yaml
+        #   (snap pack would reject them).
+        # snap.yaml can have a contact entry at the top level which is equivalent
```

Comment:
we don't support that ATM, current snapd would just ignore it, as I said new code would also ignore it

Reply:
this is from the context of snapcraft.yaml, where we never supported the contact field anyways and just a comment about an internal problem we have with transformations by handling snap.yaml and snapcraft.yaml with the same class.